buildozer/targets/android.py

```diff
@@ -99,32 +99,39 @@ def warn_on_deprecated_tokens(self):
                 self.buildozer.error(error)
 
     def _p4a(self, cmd, **kwargs):
-        if not hasattr(self, "pa_dir"):
-            self.pa_dir = join(self.buildozer.platform_dir, self.p4a_directory)
-        kwargs.setdefault('cwd', self.pa_dir)
+        kwargs.setdefault('cwd', self.p4a_dir)
         return self.buildozer.cmd(self._p4a_cmd + cmd + self.extra_p4a_args, **kwargs)
 
     @property
-    def p4a_best_android_ndk(self):
+    def p4a_dir(self):
+        """The directory where python-for-android is/will be installed."""
+
+        # Default p4a dir
+        p4a_dir = join(self.buildozer.platform_dir, self.p4a_directory_name)
+
+        # Possibly overriden by user setting
+        system_p4a_dir = self.buildozer.config.getdefault('app', 'p4a.source_dir')
+        if system_p4a_dir:
+            p4a_dir = expanduser(system_p4a_dir)
+
+        return p4a_dir
+
+    @property
+    def p4a_recommended_android_ndk(self):
         """
         Return the p4a's recommended android's NDK version, depending on the
         p4a version used for our buildozer build. In case that we don't find
         it, we will return the buildozer's recommended one, defined by global
-        variable `ANDROID_NDK_VERSION`.
+        variable `DEFAULT_ANDROID_NDK_VERSION`.
         """
-        if self.p4a_best_ndk_version is not None:
-            # make sure to read p4a version only the first time
-            return self.p4a_best_ndk_version
-
-        if not hasattr(self, "pa_dir"):
-            pa_dir = join(self.buildozer.platform_dir, self.p4a_directory)
-        else:
-            pa_dir = self.pa_dir
+        # make sure to read p4a version only the first time
+        if self.p4a_recommended_ndk_version is not None:
```

Comment:
if we implement some kind of caching, then maybe in the future we want to make it a dedicated property that automagically deals with it